plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -252,7 +279,8 @@ protected void copyTableSchema(Connection connection, String catalogName, String
                 return Optional.of(timeColumnMapping(TIME));
 
             case Types.TIMESTAMP:
-                return Optional.of(timestampColumnMappingUsingSqlTimestamp(TIMESTAMP));
+                int precision = typeHandle.getDecimalDigits().orElseThrow(() -> new IllegalStateException("timestamp precision not present"));
```

Comment:
just use `getRequiredDecimalDigits`

Reply:
Done! I also changed that for `Types.DECIMAL` for consistency.